packages/protobuf/src/create-descriptor-set.ts

```diff
@@ -793,36 +854,41 @@ function isOptionalField(
  */
 export function isPackedFieldByDefault(
   proto: FieldDescriptorProto,
-  syntax: "proto2" | "proto3",
+  syntax: "proto2" | "proto3" | "editions",
 ): boolean {
   assert(proto.type, `invalid FieldDescriptorProto: missing type`);
-  if (syntax === "proto3") {
-    switch (proto.type) {
-      case FieldDescriptorProto_Type.DOUBLE:
-      case FieldDescriptorProto_Type.FLOAT:
-      case FieldDescriptorProto_Type.INT64:
-      case FieldDescriptorProto_Type.UINT64:
-      case FieldDescriptorProto_Type.INT32:
-      case FieldDescriptorProto_Type.FIXED64:
-      case FieldDescriptorProto_Type.FIXED32:
-      case FieldDescriptorProto_Type.UINT32:
-      case FieldDescriptorProto_Type.SFIXED32:
-      case FieldDescriptorProto_Type.SFIXED64:
-      case FieldDescriptorProto_Type.SINT32:
-      case FieldDescriptorProto_Type.SINT64:
-      case FieldDescriptorProto_Type.BOOL:
-      case FieldDescriptorProto_Type.ENUM:
-        // From the proto3 language guide:
-        // > In proto3, repeated fields of scalar numeric types are packed by default.
-        // This information is incomplete - according to the conformance tests, BOOL
-        // and ENUM are packed by default as well. This means only STRING and BYTES
-        // are not packed by default, which makes sense because they are length-delimited.
-        return true;
-      default:
-        return false;
-    }
+  switch (syntax) {
+    case "proto3":
+      switch (proto.type) {
+        case FieldDescriptorProto_Type.DOUBLE:
+        case FieldDescriptorProto_Type.FLOAT:
+        case FieldDescriptorProto_Type.INT64:
+        case FieldDescriptorProto_Type.UINT64:
+        case FieldDescriptorProto_Type.INT32:
+        case FieldDescriptorProto_Type.FIXED64:
+        case FieldDescriptorProto_Type.FIXED32:
+        case FieldDescriptorProto_Type.UINT32:
+        case FieldDescriptorProto_Type.SFIXED32:
+        case FieldDescriptorProto_Type.SFIXED64:
+        case FieldDescriptorProto_Type.SINT32:
+        case FieldDescriptorProto_Type.SINT64:
+        case FieldDescriptorProto_Type.BOOL:
+        case FieldDescriptorProto_Type.ENUM:
+          // From the proto3 language guide:
+          // > In proto3, repeated fields of scalar numeric types are packed by default.
+          // This information is incomplete - according to the conformance tests, BOOL
+          // and ENUM are packed by default as well. This means only STRING and BYTES
+          // are not packed by default, which makes sense because they are length-delimited.
+          return true;
+        default:
+          return false;
+      }
+    case "proto2":
+      return false;
+    case "editions":
+      // TODO support edition featureset
+      return true;
```

Comment:
nit: I would probably unify this with "proto3" so it is at least closer to accurate. By default in editions (without element-level feature override), it behaves the same as proto3.

Reply:
In EDITION_2023 it does, but in EDITION_PROTO2 it does not 😛 Will update this logic in the next step, where this value will be plucked from FeatureSetDefaults.

Reply:
But if the edition is `EDITION_PROTO2`, the syntax will _not_ be "editions"; it will be "proto2".

Reply:
You're right about "proto2" of course. 

Here is the [new implementation of isPackedFieldByDefault](https://github.com/bufbuild/protobuf-es/pull/627/files#diff-b8e9e2ab828f5f89cf78736298aafb3fb0727af2b53730fa9ef3ba3563025348R959). It uses the seeded default value of `repeated_field_encoding` for the edition, including `EDITION_PROTO2`, `EDITION_PROTO3`, `EDITION_2023`, and any future edition.

The implementation is shorter and seems less ambiguous. I guessed it would, which is why I wanted to hold off modifying it in this PR. I should have explained better.